### PR TITLE
Review test matrix; test vs. h5py>=3.8.0

### PR DESCRIPTION
**Pull request by crusaderky**
- Increase max tested Python version from 3.11 to 3.13
- Test against pinned old versions of all dependencies
- Set the minimum required versions to the minimum tested versions
- Clean up compatibility with obsolete dependencies

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,15 +15,15 @@ authors = [
 ]
 description = "Versioned HDF5 provides a versioned abstraction on top of h5py"
 readme = "README.md"
-requires-python = ">=3.8"
+requires-python = ">=3.9"
```
Comment (crusaderky) [inline pyproject.toml:18]: 3.8 is unsupported by meson and has been untested in CI for years

File: pyproject.toml [hunk 2]
```diff
@@ -15,15 +15,15 @@ authors = [
...
 classifiers = [
     "Programming Language :: Python :: 3",
     "License :: OSI Approved :: BSD License",
     "Operating System :: OS Independent",
 ]
 dependencies = [
-    "numpy",
-    "h5py",
+    "numpy>=1.24.4",
+    "h5py>=3.3.0",
```
Comment (crusaderky) [inline pyproject.toml:26]: released June 2021

File: pyproject.toml [hunk 3]
```diff
@@ -15,15 +15,15 @@ authors = [
...
+requires-python = ">=3.9"
 classifiers = [
     "Programming Language :: Python :: 3",
     "License :: OSI Approved :: BSD License",
     "Operating System :: OS Independent",
 ]
 dependencies = [
-    "numpy",
-    "h5py",
+    "numpy>=1.24.4",
```
Comment (crusaderky) [inline pyproject.toml:25]: released June 2023

File: pyproject.toml [hunk 4]
```diff
@@ -15,15 +15,15 @@ authors = [
...
+    "h5py>=3.8.0",
```
Comment (crusaderky) [inline pyproject.toml:26]: Released January 2023

File: versioned_hdf5/backend.py [hunk 5]
```diff
@@ -438,70 +439,45 @@ def write_dataset_chunks(f, name, data_dict):
...
-    slices = {c: s.reduce() for c, s in slices.items()}
+    raw_data_name = raw_data.name.encode("utf-8")
 
     if len(raw_data) == 0:
         layout = VirtualLayout(shape=(0,), dtype=raw_data.dtype)
     else:
+        layout = VirtualLayout(shape, dtype=raw_data.dtype)
+        layout._src_filenames.add(b".")
+        space = h5s.create_simple(shape)
+        selector = Selector(space)
```
Comment (peytondmurray) [inline versioned_hdf5/backend.py:459]: :+1:

**Review by peytondmurray — APPROVED** [review]
Thanks for this :rocket:

**Pull request merged**